tests/foundry/common/abstract/type-data.mjs.test-d.ts

```diff
@@ -0,0 +1,33 @@
+import { expectTypeOf } from "vitest";
+import type DataModel from "../../../../src/foundry/common/abstract/data.d.mts";
+import type TypeDataModel from "../../../../src/foundry/common/abstract/type-data.d.mts";
+import type { fields } from "../../../../src/foundry/common/data/module.d.mts";
+import type { Merge } from "../../../../src/types/utils.d.mts";
+import type BaseJournalEntryPage from "../../../../src/foundry/common/documents/journal-entry-page.d.mts";
+
+/* attempting to use the example as a test */
+
+interface QuestSchema extends BaseJournalEntryPage.Schema {
+  description: fields.HTMLField;
+  steps: fields.ArrayField<fields.StringField>;
+}
+declare class TestModel extends TypeDataModel<
+  QuestSchema,
+  BaseJournalEntryPage,
+  { description: string; steps: string[] },
+  { totalSteps: number }
+> {}
```

Comment:
Can you change this from a simple `declare class` to a class implementation, then write tests inside overridden `prepareBaseData` and `prepareDerivedData` functions?

Reply:
I tried to but I'm still quite inexperienced in typescript. Wouldn't I have to copy paste the whole implementation of TypedDataModel from source and then inherit from that if I wanted to do what you're asking?

Reply:
No, the whole point of the test files is they're basically like writing actual end user code.